src/hotspot/share/cds/dynamicArchive.cpp

```diff
@@ -171,6 +173,28 @@ class DynamicArchiveBuilder : public ArchiveBuilder {
     FileMapInfo::metaspace_pointers_do(it);
     SystemDictionaryShared::dumptime_classes_do(it);
   }
+
+  virtual void iterate_primitive_array_klasses(MetaspaceClosure* it) {
+    for (int i = T_BOOLEAN; i <= T_LONG; i++) {
+      assert(is_java_primitive((BasicType)i), "sanity");
+      Klass* k = Universe::typeArrayKlassObj((BasicType)i);  // this give you "[I", etc
+      assert(MetaspaceShared::is_shared_static((void*)k),
+        "one-dimensional primitive array should be in static archive");
+      ArrayKlass* ak = ArrayKlass::cast(k);
+      while (ak != nullptr && ak->is_shared()) {
+        Klass* next_k = ak->array_klass_or_null();
+        if (next_k != nullptr) {
+          ak = ArrayKlass::cast(next_k);
+        } else {
+          ak = nullptr;
+        }
+      }
+      if (ak != nullptr && (ak->dimension() > 1)) {
+        // this is the lowest dimension that's not in the static archive
```

Comment:
The `ak->dimension() > 1` should either be removed, or changed into an assert.

Reply:
I changed it to an assert.